app/src/main/java/com/babylon/wallet/android/utils/ExceptionMessageProvider.kt

```diff
@@ -0,0 +1,22 @@
+package com.babylon.wallet.android.utils
+
+import android.content.Context
+import com.babylon.wallet.android.R
+import com.babylon.wallet.android.domain.asRadixWalletException
+import com.babylon.wallet.android.domain.toUserFriendlyMessage
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+
+class ExceptionMessageProvider @Inject constructor(
```

Comment:
Were is this used? Should we need it? One reason I can think of is to be used inside a view model. But context is needed in order to show something in UI. Therefore should we need this? I personally like to avoid context being used in a VM.

Reply:
there is one case in transaction status dialog where we passed UIMessage.Error to a composable for `Transaction.Status.Fail`. Since right now UiMessage.Error has only throwable inside, and throwable can't easily be passed to a composable, I've added this so we can get exception message in a view model and pass just error message string to a composable, instead of passing UIMessage.Error

Reply:
Yes I also agree here with Michael.
can't we do something like we do in `SettingsItem` with the `@StringRes` functions?
Maybe do the same with by checking the wallet error type?

Reply:
from what I understand in this refactor is that we process the whatever (radix) exception and map it to a specific wallet error type which is being used in the presentation layer.
Then I think we need to rely on this WalletErrorType to resolve the error message.

Reply:
@micbakos-rdx we already use context in View models implicitly by using DeviceSecurityHelper, FileRepository and so on

Reply:
@giannis-rdx I didn't track entire failure flow here to be sure that WalletErrorType is always non null and can be used to infer final error message. This seemed safer for the time being to grab that message in VM. Are you sure we can rely on WalletErrorType only here, or this is just suggestion?

Reply:
Based on what we have done in `SettingsItem` and from a look I had in your branch I suggested the above. Maybe `WalletErrorType` is not the class we need to rely on.

In your "error" models you pass these two properties
```
data class Fail(
   ...
   error = errorMessage, // which is a the error message from the exception
   walletErrorType = walletErrorType, // which is the mapping of the exception to wallet type
   ...
)
```
Therefore my suggestion is if we can keep only the walletErrorType and resolve the string in the compose with the same or similar way with `SettingsItem`.
Or another way could **potentially** be to only pass the `RadixException` in the data class and resolve the error message and wallet error type in composable.

If I have some time later I can try to see if the above works. Unless you think it is definitely not doable.

Reply:
the problem is that I can pass easily `RadixException` to composable, this is root cause why I did what I did here, and I didn't want to pass UiMessage.Error too, which now only has `Throwable` inside . I can take another look but it was not easy. Class that we pass to a dialog is `TransactionStatusFields`, so it's `error` field would need to be RadixWalletException, which should be `Serializable` and `Parcelable` this felt much more hassle.

Reply:
the point here is that some human friendly messages are parametrized (like WrongNetwork), this is why I return String, not IntRes, becuase I'm using data class parameters to construct parametrized string. So this is no go, I also considered it. Unless I'm missing something?

Reply:
Then I think this is the best we can do at the moment. 😄 

About the injected class I would drop it, keep he extension function, and inject the context int he delegate/viewmodel. Makes no much impact to have one more class that wraps this extension function.
Unless it helps for testing

Reply:
yes, this would help in testing so we mock that string provider instead of having context as direct dependency, although this delegate is not tested at the moment